lib/mayaUsd/ufe/UsdUndoDeleteCommand.cpp

```diff
@@ -60,20 +62,38 @@ void UsdUndoDeleteCommand::execute()
     const auto& stage = _prim.GetStage();
     auto        targetPrimSpec = stage->GetEditTarget().GetPrimSpecForScenePath(_prim.GetPath());
 
-    if (UsdUfe::applyCommandRestrictionNoThrow(_prim, "delete")) {
+    PXR_NS::UsdEditTarget routingEditTarget
+        = getEditRouterEditTarget(UsdUfe::EditRoutingTokens->RouteDelete, _prim);
+
+    if (!UsdUfe::applyCommandRestrictionNoThrow(_prim, "delete", true))
+        return;
+
 #ifdef UFE_V4_FEATURES_AVAILABLE
-        UsdAttributes::removeAttributesConnections(_prim);
+    UsdAttributes::removeAttributesConnections(_prim);
 #endif
-        // Let removeAttributesConnections be run first as it will also cleanup
-        // attributes that were authored only to be the destination of a connection.
-        if (!UsdUfe::cleanReferencedPath(_prim)) {
-            const std::string error = TfStringPrintf(
-                "Failed to cleanup references to prim \"%s\".", _prim.GetPath().GetText());
+    // Let removeAttributesConnections be run first as it will also cleanup
+    // attributes that were authored only to be the destination of a connection.
+    if (!UsdUfe::cleanReferencedPath(_prim)) {
+        const std::string error = TfStringPrintf(
+            "Failed to cleanup references to prim \"%s\".", _prim.GetPath().GetText());
+        TF_WARN("%s", error.c_str());
+        throw std::runtime_error(error);
+    }
+
+    if (!routingEditTarget.IsNull()) {
+        PXR_NS::UsdEditContext ctx(stage, routingEditTarget);
```

Comment:
The edit context to change the edit target should be set before calling `applyCommandRestrictionNoThrow()` above, since that function verifies that the edit can be made for the current target. Otherwise the check and the action are not in sync.

Reply:
Sure thing.